File: src/types/credential.ts
export interface Credential {
  id: string;
  label: string;
  platform: string;
  cookie_path: string;
  is_default: boolean;
  created_at: string;
  updated_at: string;
}

export interface CreateCredentialRequest {
  label: string;
  platform: string;
  cookie_path: string;
  is_default?: boolean;
}

export interface UpdateCredentialRequest {
  label?: string;
  cookie_path?: string;
  is_default?: boolean;
}

File: src-tauri/src/models/credential.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct Credential {
    pub id: String,
    pub label: String,
    pub platform: String,
    pub cookie_path: String,
    pub is_default: bool,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Deserialize)]
pub struct CreateCredentialRequest {
    pub label: String,
    pub platform: String,
    pub cookie_path: String,
    pub is_default: Option<bool>,
}

#[derive(Debug, Deserialize)]
pub struct UpdateCredentialRequest {
    pub label: Option<String>,
    pub cookie_path: Option<String>,
    pub is_default: Option<bool>,
}
